connection-api/src/main/java/org/terracotta/connection/entity/EntityRef.java

```diff
@@ -56,34 +77,33 @@
   C reconfigure(C configuration) throws EntityException;
 
   /**
-   * Destroy the entity pointed to by this reference.
-   * Note that this will block if there are any open instances of the fetched entity on any client.
-   * 
-   * @throws EntityNotProvidedException The service providing T doesn't exist on either the client or the server
-   * @throws EntityNotFoundException No entity with this type and name could be found
-   */
-  void destroy() throws EntityNotProvidedException, EntityNotFoundException;
+   * Destroys the entity pointed to by this reference if it is not fetched on any client.
+   * <P>
+   *   If the entity is fetched on any client, this method returns false.
+   * </P>
+   *
+   * @return {@code true} if the entity was destroyed, {@code false} if there are fetched entities on any client
+   * preventing destruction
```

Comment:
Given that we are building mostly on top of non blocking ops, I think the defaults should have the same characteristics. Now we could use an exception ... but that seems weird in some ways.